di_art/lib/parameter/definition_parameters.dart

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 DiArt.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:di_art/error/NoParameterFoundException.dart';
+import 'package:logart/logart.dart';
+
+///
+/// DefinitionParameters - Parameter holder
+/// Usable with exploded declaration
+///
+class DefinitionParameters {
+  List params;
+
+  DefinitionParameters(this.params);
```

Comment:
How about such constructor:
```
DefinitionParameters([this.params = const[]]);
```
then we can change `DefinitionParameters emptyParametersHolder() => DefinitionParameters([]);` to `DefinitionParameters emptyParametersHolder() => DefinitionParameters();`